### PR TITLE
API trace_call implementation

### PR DESCRIPTION
**Pull request by Sixtysixter**
Implementation of trace_call API

### CODE REVIEW

File: silkrpc/commands/eth_api.cpp [hunk 1]
```diff
@@ -1099,12 +1099,16 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_create_access_list(const nlohma
         }
 
         auto tracer = std::make_shared<AccessListTracer>(*call.from, to);
+
+        silkrpc::Tracers tracers;
+        tracers.push_back(tracer);
```
Comment (canepat) [inline silkrpc/commands/eth_api.cpp:1104]: This can be simplified: `Tracers call_tracers{tracer};`

File: silkrpc/commands/trace_api.cpp [hunk 2]
```diff
@@ -17,25 +17,65 @@
...
+        }
+        if (entry == "vmTrace") {
+            config.vm_trace = true;
+        }
+        if (entry == "stateDiff") {
+            config.state_diff = true;
+        }
+    }
+
+    SILKRPC_LOG << "call: " << call << " block_number_or_hash: " << block_number_or_hash << " config: " << config << "\n";
```
Comment (canepat) [inline silkrpc/commands/trace_api.cpp:62]: You should use a lower log level here

File: silkrpc/core/evm_debug.cpp [hunk 3]
```diff
@@ -268,7 +269,10 @@ asio::awaitable<std::vector<DebugTrace>> DebugExecutor<WorldState, VM>::execute(
         debug_trace.debug_config = config_;
         auto debug_tracer = std::make_shared<debug::DebugTracer>(debug_trace.debug_logs, config_);
 
-        const auto execution_result = co_await executor.call(block, txn, /* refund */false, /* gasBailout */false, debug_tracer);
+        silkrpc::Tracers tracers;
+        tracers.push_back(debug_tracer);
```
Comment (canepat) [inline silkrpc/core/evm_debug.cpp:273]: Probably better this way: `Tracers debug_tracers{debug_tracer};`

File: silkrpc/core/evm_debug.cpp [hunk 4]
```diff
@@ -293,7 +297,7 @@ asio::awaitable<DebugExecutorResult> DebugExecutor<WorldState, VM>::execute(cons
 template<typename WorldState, typename VM>
 asio::awaitable<DebugExecutorResult> DebugExecutor<WorldState, VM>::execute(std::uint64_t block_number, const silkworm::Block& block,
         const silkrpc::Transaction& transaction, std::int32_t index) {
-    SILKRPC_DEBUG << "execute: "
+    SILKRPC_LOG << "execute: "
```
Comment (canepat) [inline silkrpc/core/evm_debug.cpp:300]: You should use a lower log level here

File: silkrpc/core/evm_debug.cpp [hunk 5]
```diff
@@ -321,10 +325,14 @@ asio::awaitable<DebugExecutorResult> DebugExecutor<WorldState, VM>::execute(std:
     debug_trace.debug_config = config_;
 
     auto debug_tracer = std::make_shared<debug::DebugTracer>(debug_trace.debug_logs, config_);
-    const auto execution_result = co_await executor.call(block, transaction, /* refund */ false, /* gasBailout*/ false, debug_tracer);
+
+    silkrpc::Tracers tracers;
+    tracers.push_back(debug_tracer);
```
Comment (canepat) [inline silkrpc/core/evm_debug.cpp:330]: See comment above

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkrpc/core/evm_debug.cpp (hunk 3 above)
Comment (Sixtysixter) [reply silkrpc/core/evm_debug.cpp:273]: Done

File: silkrpc/core/evm_debug.cpp (hunk 4 above)
Comment (Sixtysixter) [reply silkrpc/core/evm_debug.cpp:300]: Log level set to INFO

File: silkrpc/core/evm_debug.cpp (hunk 5 above)
Comment (Sixtysixter) [reply silkrpc/core/evm_debug.cpp:330]: Done

File: silkrpc/commands/trace_api.cpp (hunk 2 above)
Comment (Sixtysixter) [reply silkrpc/commands/trace_api.cpp:62]: Done

**Review by canepat — APPROVED** [review]

**Pull request merged**